siac/rentercmd.go

```diff
@@ -381,37 +388,7 @@ func renterfilesdeletecmd(path string) {
 func renterfilesdownloadcmd(path, destination string) {
 	destination = abs(destination)
 	done := make(chan struct{})
-	go func() {
-		time.Sleep(time.Second) // give download time to initialize
-		for {
-			select {
-			case <-done:
-				return
-
-			case <-time.Tick(time.Second):
-				// get download progress of file
-				var queue api.RenterDownloadQueue
-				err := getAPI("/renter/downloads", &queue)
-				if err != nil {
-					continue // benign
-				}
-				var d modules.DownloadInfo
-				for _, d = range queue.Downloads {
-					if d.Destination == destination {
-						break
-					}
-				}
-				if d.Filesize == 0 {
-					continue // file hasn't appeared in queue yet
-				}
-				pct := 100 * float64(d.Received) / float64(d.Filesize)
-				elapsed := time.Since(d.StartTime)
-				elapsed -= elapsed % time.Second // round to nearest second
-				mbps := (float64(d.Received*8) / 1e6) / time.Since(d.StartTime).Seconds()
-				fmt.Printf("\rDownloading... %5.1f%% of %v, %v elapsed, %.2f Mbps    ", pct, filesizeUnits(int64(d.Filesize)), elapsed, mbps)
-			}
-		}
-	}()
+	go downloadprogress(done, path)
 
 	err := get("/renter/download/" + path + "?destination=" + destination)
```

Comment:
I don't see a reason for it to be, we want this command to block until the download is finished, using downloadasync here would just introduce a bunch of ugly polling